src/modules/kvs-watch/kvs-watch.c

```diff
@@ -209,19 +209,13 @@ static struct ns_monitor *namespace_create (struct watch_ctx *ctx,
     return NULL;
```

Comment:
In 577ba928723fc73bff3b8d685df616550a99cfc7 commit message:

> In the function, they array of keys is regularly searched to determine if any keys are duplicates of each other.
This is is an O(N^2) search.

s/they/the/



Wouldn't it be O(N)?

Reply:
I think O(n^2).  I believe what was going on was equivalent to:

```
for (i = 0; i < n; i++) {
   for (j = i + 1; j < n; j++) {
      if str[i] == str[j]
          there's a duplicate
          break
   }
}
```

Reply:
Actually I was wondering if `strcmp()` implementations are a lot better than that these days. When I was looking at `perf` results I saw calls to `__strcmp_sse2_*()` instead of just `strcmp()`. I wonder if strings can be compared with SSE2 instructions in better than O(N)? I'd be interested to know the answer to this because I have no idea.